### PR TITLE
[SparkR][Minor] Fix doc for show method

### PR DESCRIPTION
**Pull request by junyangq**
## What changes were proposed in this pull request?

The original doc of `show` put methods for multiple classes together but the text only talks about `SparkDataFrame`. This PR tries to fix this problem.
## How was this patch tested?

Manual test.

### CODE REVIEW

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -212,9 +212,9 @@ setMethod("showDF",
 
 #' show
 #'
-#' Print the SparkDataFrame column names and types
+#' Print class and type information of a SparkR object.
```
Comment (junyangq) [inline R/pkg/R/DataFrame.R]: not sure if there is a better name for this collection

Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: we don't really say "SparkR object", I think. How about "Spark object"?

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -212,9 +212,9 @@ setMethod("showDF",
...
 #'
-#' @param object a SparkDataFrame.
+#' @param object a SparkR object. Can be a SparkDataFrame, Column, GroupedData, WindowSpec.
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: ditto "Spark object".
in that case, let's move this @param to generics.R like we discussed?

File: R/pkg/R/DataFrame.R (hunk 1 above)
Comment (junyangq) [inline R/pkg/R/DataFrame.R]: Ok sounds good to me.

File: R/pkg/R/DataFrame.R (hunk 2 above)
Comment (junyangq) [inline R/pkg/R/DataFrame.R]: It's not in the generic function definition. If we add that, then it seems `method::show` will be masked.

Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: ok this is the case the existing generic from method is used.
let's leave it then.

**Comment by felixcheung** [comment]
LGTM. waiting for Jenkins.

**Pull request closed without merging**